src/network/gossip.rs

```diff
@@ -188,29 +189,20 @@ impl SnapchainGossip {
             }
         }
 
-        if read_node {
-            let topic = gossipsub::IdentTopic::new(READ_NODE_TOPIC);
-            let result = swarm.behaviour_mut().gossipsub.subscribe(&topic);
-            if let Err(e) = result {
-                warn!("Failed to subscribe to topic: {:?}", e);
-                return Err(Box::new(e));
-            }
-        } else {
-            // Create a Gossipsub topic
-            let topic = gossipsub::IdentTopic::new(CONSENSUS_TOPIC);
-            // subscribes to our topic
-            let result = swarm.behaviour_mut().gossipsub.subscribe(&topic);
-            if let Err(e) = result {
-                warn!("Failed to subscribe to topic: {:?}", e);
-                return Err(Box::new(e));
-            }
+        // Create a Gossipsub topic
+        let topic = gossipsub::IdentTopic::new(CONSENSUS_TOPIC);
```

Comment:
I think we still want the if check? The read nodes shouldn't subscribe to the consensus topic